mixnet/src/topology.rs

```diff
@@ -69,3 +92,105 @@ impl From<MixNodeInfo> for sphinx_packet::route::Destination {
         )
     }
 }
+
+impl Serialize for MixNodeInfo {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        SerializableMixNodeInfo::try_from(self)
+            .unwrap()
+            .serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for MixNodeInfo {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Ok(Self::try_from(SerializableMixNodeInfo::deserialize(deserializer)?).unwrap())
+    }
+}
```

Comment:
I'm pretty sure you actually would like to use [`with`](https://serde.rs/remote-derive.html)

```rust
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct MixNodeInfo(#[serde(with = "SerializableMixNodeInfo")] sphinx_packet::route::Node);
```

Reply:
Thanks. I just tried to use `#[serde(with = "..")]` but failed because of errors that are returned from `try_from`s. Let me try it again.